src/python-frontend/python_converter.cpp

```diff
@@ -803,12 +803,20 @@ exprt python_converter::get_function_call(const nlohmann::json &element)
        * For example, x = int(1) becomes x = 1. */
       size_t arg_size = 1;
       const auto &arg = element["args"][0];
+	
+      nlohmann::json temp_arg = arg;
 
+      if (func_name == "int" && arg["value"].is_number())
+      {
+        double arg_value = arg["value"].get<double>();
+        temp_arg["value"] = static_cast<int>(arg_value);  // Truncate value in the copy
+      }
```

Comment:
Some suggestions:
- Remove `const` from `arg` and manipulate it directly.
- Add it as an `else` block for the `if (func_name == "str")`
- Use `arg["value"].is_number_float()` instead of `arg["value"].is_number()`